articles/best-practices/rules.md

```diff
@@ -1,129 +1,571 @@
 ---
 description: Best practices for writing and managing Auth0 Rules.
+toc: true
 topics:
   - best-practices
   - rules
-contentType:
-    - reference
+contentType: reference
 useCase:
   - best-practices
   - rules
 ---
 
 # Rules Best Practices
 
-This article covers best practices for using [rules](/rules). Before you start writing rules, we recommend that you:
+This article covers some best practices when using [rules](/rules). Rules can be used in a [variety of situations](/rules#what-can-i-use-rules-for-) as part of the pipeline where artifacts for authenticity are generated - i.e. an [ID Token](/tokens/id-token) in [OpenID Connect (OIDC)](/protocols/oidc), an [Access Token](/tokens/overview-access-tokens) in [OAuth 2.0](/protocols/oauth2), or an [assertion in SAML](/protocols/saml/saml-configuration/saml-assertions#use-rules). A new pipeline is created for each authentication request, in which rules execute.
```

Comment:
Too many links in this paragraph make it hard to read. Maybe change the OIDC and Access Token links to glossary term links instead. Also, this one (/rules#what-can-i-use-rules-for-) ends in a dash. Not sure if it's a typo.

Glossary links to replace other links: 

- `<dfn data-key="access-token">Access Token</dfn>`
- `<dfn data-key="openid">OpenID Connect (OIDC)</dfn>`

Reply:
Re-factored to use glossary reference for Access Token. It would be useful to have a glossary reference for ID Token too - which currently seems to be missing :-(

Hyperlink format (with dash at end) is correct; good observation though, thank you :-)